### ISSUE TITLE
Advanced hunger bar

### ISSUE THREAD

**Issue by BenCheung0422** [MinicraftPlus/minicraft-plus-revived#713]
## Background

Currently, the hunger system is simply just consuming hunger points by exhaustion time and whether decrementing consuming exhaustion time each tick. The system is quite unflexible when there are several actions that would "consume" exhaustion time in different extents, since it is difficult to set explicit values. Now they are mainly done by checking whether the current game time is a factor of 2 or random of a range equals to zero or such.

## Suggestions

By considering mature system like the one in Minecraft, [the hunger system](https://minecraft.wiki/w/Hunger), the system can be expanded and standardized in a better way.

### Exhaustion Levels

Firstly, turning the exhaustion time into exhaustion levels. The exhaustion level is checked and calculated each tick, so everything can be covered at once. Each actions and conditions would give different exhaustion levels; per a certain value of exhaustion levels, the hunger level decrements. This way, it can make sure that hunger exhaustion would not be too complicated or verbose, but instead more standardized and simple.

### Food Saturation Level

Secondly, other than just the visual hunger points, there can be food saturation level. By eating food, the saturation level increases. In hunger level decrement, the saturation level is consumed before the hunger points. This can strengthen the effect of food saturation at the same time reduce the food demand. It is not necessary to visualize this value in the HUD like Apple Skin.

**Comment by zandgall** [comment]
I do not think we should implement this, it is both taking too much influence by directly copying minecraft's hunger system, and also neglects a more RPG-oriented direction I would rather see this project move in.
